### ISSUE TITLE
Add links to contribution guidelines using Githubs infrastructure

### ISSUE THREAD

**Issue by gavanderhoorn** [ros-industrial/ros_industrial_issues#26]
Github parses `CONTRIBUTING(.md)` files in repositories and displays a link to such a file at the top of all pull requests and the _new issue_ page (see [[1]](https://github.com/blog/1184-contributing-guidelines)).

We could use this to direct people to the ROS-I coding and pull request standards / guidelines.

**Comment by gavanderhoorn** [comment]
Related: #39.

**Comment by tfoote** [comment]
I've opened PRs to add or update CONTRIBUTING.md to all the repos that have been recently touched.

**Comment by tfoote** [comment]
A good suggestion that this should be generally updated to mention the new Github review process: https://github.com/ros-industrial/abb/pull/153#pullrequestreview-136511382

**Comment by tfoote** [comment]
The content I submitted was: https://gist.github.com/tfoote/9f38576e148c82f86aa7e6871104a77f

It's a markdown version of https://rosindustrial.org/developmentprocess/

I'm not going to have time to follow up on most of these after wrid18.

**Comment by gavanderhoorn** [comment]
everyone in ROS-I: could you hold off on merging the PRs submitted by @tfoote? As commented in https://github.com/ros-industrial/abb/pull/153#pullrequestreview-136511382, the PRs are based on the `Development Process` page, but that hasn't been updated to use the 'new' Github review system.

**Comment by shaun-edwards** [comment]
I'm inclined to accept the PRs as is.  The steps are in spirit of Github AND having a CONTRIBUTING file is better than not having it.  It looks like @Levi-Armstrong has already moved forward with @tfoote's version.

**Comment by gavanderhoorn** [comment]
Yes, but we've also received comments by community members (such as those in https://github.com/ros-industrial/abb/pull/153#pullrequestreview-136511382) that are valid.

And I don't necessarily agree that having a `CONTRIBUTING.md` is better than not having one: having a process is important. The file itself is not.

**Comment by mathias-luedtke** [comment]
It might have been better to agree one a consolidated version beforehand in a **central** place.
(Same story as with #55)

We could discuss an improved version now, however not spread in 20+ PRs.
I would suggest to open one PR in a (to be created?) repository with common templates etc.
After a reasonable amount of time (2 weeks?) we could update the other PRs based on this new template.

**Comment by tfoote** [comment]
A draft was precirculated in the wrid2018 IRC prior to posting. But no one caught that the last paragraph was out of date.
Note that the file itself is somewhat important as Github uses it in it's user interaction prompts which will help new and learning contributors.

**Comment by gavanderhoorn** [comment]
@ipa-mdl wrote:
> I would suggest to open one PR in a (to be created?) repository with common templates etc.

Would that be something you would be willing to do @ipa-mdl?

I can create the repository for you.

**Comment by gavanderhoorn** [comment]
Unassigning @tfoote as it's no longer his task.

Thanks @tfoote for all the PRs.

We'll need to take a good look at this during wrid19, so marking this issue as one for that day.
